libs/build_esmf.sh

```diff
@@ -110,7 +110,8 @@ case $MPI in
     export ESMF_COMM=${STACK_esmf_comm:-"mpich3"}
     ;;
   cray-mpich )
-    export ESMF_COMM=${STACK_esmf_comm:-"mpi"}
+    export ESMF_OS="Linux"
+    export ESMF_COMM=${STACK_esmf_comm:-"mpich3"}
```

Comment:
@climbfuji 
You may need to check how this affects gaea.

Reply:
Thanks for letting me know, I doubt this will work but I'll give it a try with this branch.

Reply:
No, this does not work on gaea, we'll need to find another solution for wcoss2.
```
make[5]: Leaving directory '/lustre/f2/scratch/esrl/Dom.Heinzeller/hpc-stack-test/hpc-stack/pkg/ESMF_8_1_0_beta_snapshot_27/src/apps/ESMF_InfoC'
CC   -m64 -mcmodel=small -pthread -Wl,--no-as-needed  -qopenmp -L/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/hdf5/1.10.6/lib   -L/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/zlib/1.2.11/lib -L/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/esmf/8_1_0_beta_snapshot_27/lib -L/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/netcdf/4.7.4/lib -L/opt/intel/compilers_and_libraries_2018.6.288/linux/compiler/lib/intel64_lin/ -Wl,-rpath,/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/esmf/8_1_0_beta_snapshot_27/lib -Wl,-rpath,/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/netcdf/4.7.4/lib -Wl,-rpath,/opt/intel/compilers_and_libraries_2018.6.288/linux/compiler/lib/intel64_lin/ -o /lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/esmf/8_1_0_beta_snapshot_27/bin/ESMF_InfoC /lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack/pkg/ESMF_8_1_0_beta_snapshot_27/obj/objO/Linux.intel.64.mpich3.default/src/apps/ESMF_InfoC/ESMF_InfoC.o -lesmf  -lmpifort -ldarshan -lfmpich -ldarshan -lz -lAtpSigHandler -lAtpSigHCommData -lpthread -lrt -lugni -lpthread -lpmi -limf -lm -ldl -lsci_intel_mpi_mp -limf -lm -lsci_intel_mp -ldl -lrt -lugni -lpthread -lpmi -limf -lm -ldl -lpmi -lpthread -lalpslli -lpthread -lwlm_detect -lalpsutil -lpthread -lrca -lxpmem -lugni -lpthread -ludreg -lsci_intel_mp -limf -lm -lpthread -liomp5 -ldl -lhugetlbfs -limf -lm -lpthread -lifport -lifcore_pic -limf -lsvml -lm -lipgo -liomp5 -lirc -lpthread -lsvml -ldl -lgcc -lgcc_eh -lirc_s -ldl -lrt -ldl -lnetcdff -lnetcdf -lhdf5_hl -lhdf5  -lz -ldl -lm
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lmpifort
make[4]: *** [/lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack/pkg/ESMF_8_1_0_beta_snapshot_27/build/common.mk:2426: /lustre/f2/scratch/Dom.Heinzeller/hpc-stack-test/hpc-stack-install/intel-18.0.6.288/cray-mpich-7.7.11/esmf/8_1_0_beta_snapshot_27/bin/ESMF_InfoC] Error 1
```

Reply:
Thanks for checking @climbfuji 
There are 2 options.
1. in `stack_wcoss2.yaml` one can define `comm=mpich3` in the `esmf` section of the yaml.
2. Eliminate `stack_wcoss2.yaml` and use `stack_noaa.yaml`.  In `config_wcoss2.sh`, put `export STACK_esmf_comm="mpich3"`.

I would prefer 2 since it consolidates the yaml files for all NOAA machines.  We anyways have to maintain a separate `config.sh` file for each platform due to compiler/mpi changes.
Thoughts?

Reply:
I would prefer 2 as well.

Reply:
@climbfuji Can you confirm these changes work on Gaea.
I would like to merge this branch before the EOTW.